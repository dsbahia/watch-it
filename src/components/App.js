import React, { useState } from "react";
import { Routes, Route } from "react-router-dom";
import { Toaster } from "react-hot-toast";
import SearchBar from "./SearchBar";
import NavBar from "./NavBar";
import UpComing from "./UpComing";
import TrendingMovieContainer from "./TrendingMoviesContainer";
import watchItLogo from "../images/watch-it.png";
import SearchResultsCard from "./SearchResultsCard";
import Footer from "./Footer";
import "../styles/App.css";

function App() {
  const [searchResults, setSearchResults] = useState({});

  return (
    <div className="App">
      <Toaster />
      <NavBar />
      <img className="watch-it-logo" src={watchItLogo} alt="Watch It Logo" />
      <SearchBar setSearchResults={setSearchResults} />{" "}
      <Routes>
        <Route path="/" element={<UpComing />} />
      </Routes>
      <TrendingMovieContainer />
      <SearchResultsCard results={searchResults.results} />
      <Footer />
    </div>
  );
}

export default App;
