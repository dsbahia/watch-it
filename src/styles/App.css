body {
  background-color: #EEEEEE;
  font-family: 'Roboto', sans-serif;
}

.App {
  text-align: center;
}

.watch-it-logo {
  width: 250px;
  height: auto;
}
