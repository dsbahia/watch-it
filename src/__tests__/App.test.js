import { render, screen } from "@testing-library/react";
import App from "../components/App";

xtest("renders learn react link", () => {
  render(<App />);
  const linkElement = screen.getByText(/Hello World!/i);
  expect(linkElement).toBeInTheDocument();
});
