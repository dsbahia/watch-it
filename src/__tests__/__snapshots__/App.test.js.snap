// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      class="navbar"
    >
      <div
        class="navbar-links-item"
      />
      <a
        class="item-upcoming"
        href="/"
      >
        Upcoming
      </a>
      <a
        class="item-trending"
        href="/"
      >
        Trending
      </a>
    </div>
    <div
      class="upcoming"
    />
    <img
      alt="Watch It Logo"
      class="watch-it-logo"
      src="watch-it.png"
    />
    <div
      class="search-bar"
    >
      <form
        class="search-form"
      >
        <input
          class="search-input"
          data-testid="search-input"
          placeholder="Search"
          type="text"
        />
        <button
          class="search-button"
          type="submit"
        >
          Go!
        </button>
      </form>
    </div>
    <footer>
      <div
        class="footer-content"
      >
        <div
          class="footer-logo"
        >
          <img
            alt="TMDB logo"
            src="TMDB icon.jpeg"
          />
        </div>
        <div
          class="footer-links"
        >
          <a
            href="https://www.themoviedb.org/about"
          >
            About TMDB
          </a>
          <a
            href="https://developer.themoviedb.org/docs"
          >
            Getting started
          </a>
          <a
            href="https://github.com/dsbahia/watch-it"
          >
            Design Team Github
          </a>
        </div>
      </div>
      <p
        class="tmdb-info"
      >
        Acknowledgements to The Movie Database for the use of the API
      </p>
      <p
        class="team-info"
      >
        Design Team: Dal Bahia, Max Staite and Jonathan Thompson
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
